### ISSUE TITLE
Uncaught TypeError: Invalid Point: (NaN, 0)

### ISSUE THREAD

**Issue by trusktr** [atom/atom#12081]
[Enter steps to reproduce below:]
1. Not sure how to reproduce. Just made split panes and eventually it happened.

**Atom Version**: 1.8.0
**System**: Mac OS X 10.10.2
**Thrown From**: Atom Core
### Stack Trace

Uncaught TypeError: Invalid Point: (NaN, 0)

```
At /opt/homebrew-cask/Caskroom/atom/1.6.1/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/point.js:63

TypeError: Invalid Point: (NaN, 0)
    at Function.module.exports.Point.assertValid (/opt/homebrew-cask/Caskroom/atom/1.6.1/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/point.js:63:15)
    at TextBuffer.module.exports.TextBuffer.clipPosition (/opt/homebrew-cask/Caskroom/atom/1.6.1/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:1061:13)
    at DisplayBuffer.module.exports.DisplayBuffer.screenPositionForBufferPosition (/opt/homebrew-cask/Caskroom/atom/1.6.1/Atom.app/Contents/Resources/app.asar/src/display-buffer.js:796:27)
    at DisplayBuffer.module.exports.DisplayBuffer.screenRangeForBufferRange (/opt/homebrew-cask/Caskroom/atom/1.6.1/Atom.app/Contents/Resources/app.asar/src/display-buffer.js:758:20)
    at TextEditor.module.exports.TextEditor.screenRangeForBufferRange (/opt/homebrew-cask/Caskroom/atom/1.6.1/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1371:33)
    at TextEditorComponent.module.exports.TextEditorComponent.onGutterClick (/opt/homebrew-cask/Caskroom/atom/1.6.1/Atom.app/Contents/Resources/app.asar/src/text-editor-component.js:801:40)
    at TextEditorComponent.onGutterClick (/opt/homebrew-cask/Caskroom/atom/1.6.1/Atom.app/Contents/Resources/app.asar/src/text-editor-component.js:3:61)
    at TextEditorComponent.module.exports.TextEditorComponent.onLineNumberGutterMouseDown (/opt/homebrew-cask/Caskroom/atom/1.6.1/Atom.app/Contents/Resources/app.asar/src/text-editor-component.js:793:21)
    at LineNumberGutterComponent.<anonymous> (/opt/homebrew-cask/Caskroom/atom/1.6.1/Atom.app/Contents/Resources/app.asar/src/text-editor-component.js:3:61)
    at HTMLDivElement.<anonymous> (/opt/homebrew-cask/Caskroom/atom/1.6.1/Atom.app/Contents/Resources/app.asar/src/line-number-gutter-component.js:3:61)
```
### Commands

```
     -1:00.4.0 symbols-list:toggle (atom-text-editor.editor.vim-mode.normal-mode.is-focused)
     -0:25.5.0 vim-mode:move-down (atom-text-editor.editor.vim-mode.normal-mode.is-focused)
     -0:25.4.0 vim-mode:move-up (atom-text-editor.editor.vim-mode.normal-mode.is-focused)
     -0:25.3.0 vim-mode:move-down (atom-text-editor.editor.vim-mode.normal-mode.is-focused)
     -0:25.2.0 vim-mode:move-up (atom-text-editor.editor.vim-mode.normal-mode.is-focused)
     -0:25.1.0 vim-mode:move-down (atom-text-editor.editor.vim-mode.normal-mode.is-focused)
     -0:25 vim-mode:move-up (atom-text-editor.editor.vim-mode.normal-mode.is-focused)
     -0:24.8.0 vim-mode:move-down (atom-text-editor.editor.vim-mode.normal-mode.is-focused)
     -0:24.8.0 vim-mode:move-up (atom-text-editor.editor.vim-mode.normal-mode.is-focused)
     -0:24.7.0 vim-mode:move-down (atom-text-editor.editor.vim-mode.normal-mode.is-focused)
     -0:24.6.0 vim-mode:move-up (atom-text-editor.editor.vim-mode.normal-mode.is-focused)
     -0:24.5.0 vim-mode:move-down (atom-text-editor.editor.vim-mode.normal-mode.is-focused)
     -0:24.5.0 vim-mode:move-up (atom-text-editor.editor.vim-mode.normal-mode.is-focused)
     -0:24.4.0 vim-mode:move-down (atom-text-editor.editor.vim-mode.normal-mode.is-focused)
  7x -0:24.1.0 vim-mode:move-up (atom-text-editor.editor.vim-mode.normal-mode.is-focused)
  7x -0:23 vim-mode:move-down (atom-text-editor.editor.vim-mode.normal-mode.is-focused)
```
### Config

``` json
{
  "core": {
    "allowPendingPaneItems": false,
    "disabledPackages": [
      "vim-mode-plus-move-to-symbols",
      "vim-mode-plus-move-selected-text",
      "vim-mode-plus-project-find-from-search",
      "todo-show",
      "language-yaml",
      "language-toml",
      "language-xml",
      "language-c",
      "language-clojure",
      "language-csharp",
      "language-go",
      "language-java",
      "language-make",
      "language-objective-c",
      "language-perl",
      "language-php",
      "language-python",
      "language-ruby",
      "language-ruby-on-rails",
      "language-sql",
      "welcome",
      "archive-view",
      "vim-mode-plus-ex-mode",
      "vim-mode-plus",
      "pigments",
      "minimap-pigments",
      "language-javascript-jsx",
      "language-jsx",
      "language-babel",
      "term3",
      "floobits"
    ],
    "ignoredNames": [
      ".git",
      ".hg",
      ".svn",
      ".DS_Store",
      "._*",
      "Thumbs.db",
      "node_modules"
    ],
    "packagesWithKeymapsDisabled": []
  },
  "editor": {
    "invisibles": {},
    "scrollPastEnd": true,
    "showIndentGuide": true,
    "tabLength": 4
  }
}
```
### Installed Packages

``` coffee
# User
auto-detect-indentation, v1.1.0
autoclose-html, v0.23.0
bottom-dock, v0.4.4
clipboard-plus, v0.5.1
ex-mode-sort, v1.0.0
file-icons, v1.7.14
glsl-preview, v0.8.0
highlight-selected, v0.11.2
language-glsl, v2.0.1
language-viml, v1.0.0
linter, v1.11.8
linter-coffeelint, v1.1.2
linter-eslint, v7.2.2
linter-jsonlint, v1.2.5
minimap, v4.24.4
minimap-bookmarks, v0.3.1
minimap-codeglance, v0.4.5
minimap-cursorline, v0.2.0
minimap-find-and-replace, v4.5.1
minimap-git-diff, v4.3.1
minimap-highlight-selected, v4.4.0
minimap-linter, v1.1.1
minimap-split-diff, v0.3.1
nerd-treeview, v0.4.2
persistent-undo, v1.2.0
project-plus, v0.9.0
quick-highlight, v0.5.0
react, v0.15.0
relative-numbers, v0.5.1
remember-file-positions, v0.2.2
symbols-list, v2.3.0
terminal-plus, v0.14.5
todo-manager, v0.2.9
tool-bar, v0.4.0
tool-bar-main, v0.0.9
tree-view-open-files, v0.3.0
vim-mode, v0.65.1
vim-mode-clipboard-plus, v0.2.0
vim-mode-visual-block, v0.2.15
vim-surround, v0.8.1

# Dev
No dev packages
```

**Comment by trusktr** [comment]
Screenshot shows pane on top left does not show any text content, and cannot scroll to it:

![screen shot 2016-07-01 at 4 55 45 am](https://cloud.githubusercontent.com/assets/297678/16520886/18b03650-3f48-11e6-958c-a56e8daa7e4e.png)

**Comment by trusktr** [comment]
The problem persists until I close the affected tabs (in this case, both tabs in the top left pane) and re-open the files.

**Comment by lee-dohm** [comment]
How did you split panes? Is this only happening in one project? Or all projects?

**Comment by trusktr** [comment]
I'm not sure how it was caused. I'll try to pay attention next time it happens, if it happens.

**Comment by brumm** [comment]
I'm encountering this with some regularity, and I _think_ it usually involves split panes, which I split by dragging a tab and invoking the animated pane split overlay.
It also seems to happen when coming back to a previously inactive tab.

@trusktr Are you using https://github.com/t9md/atom-narrow by any chance?

**Comment by t9md** [comment]
I could 100% reproduce and fixed in following commit.
Why I share this is this issue is NOT directly related to atom-narrow package, so likely to happen in other package which is calling `editorElement.setScrollTop`.

https://github.com/t9md/atom-narrow/commit/1009bc74e1d1a2be914dfff60513994093a6b646

As I wrote in comment in above commit.

When trying to restore saved scrollTop, immediately setting scrollTop after changing active-pane-item cause editor blank.
Detailed condition is described t9md/atom-narrow#95.

Here is explanatory code to describe how I work-around this.

```coffeescript
  editorElement = editor
  # Save scrollTop to restore afterward
  savedScrollTop = editorElement.getScrollTop()

  # Do something which change active pane item to different item from original editor
  pane.activateItem(differentEditor)
  # (snip) do something
  pane.activateItem(editor)

  # immediate calling editorElement.scrollTop after changing activeItem cause editor content blank.
  # See detailed condition this happens in atom-narrow#95.
  # In this state, component.getScrollTop() returns `undefined`, in this case delaying setScrollTop.
  unless editorElement.component.getScrollTop()?
    disposable = editorElement.onDidChangeScrollTop ->
      disposable.dispose()
      editorElement.setScrollTop(savedScrollTop)
  else
    editorElement.setScrollTop(savedScrollTop)
```

**Comment by t9md** [comment]
I'll put minimum repro code here later within hour.

**Comment by t9md** [comment]
Here is GIF when I repro using command in following gist.
What I did in GIF is
1. open init.coffee
2. invoke `user:repro`, (pane split-righted and right-pane is activated)
3. close right pane, (active pane is now left pane, active editor is init.coffee )
4. split-right blank out init.coffee at left-pane, clicking it throw error.

https://gist.github.com/t9md/557babb6f17de583ec8c5e8f0cabd1cb

![reproduce-atom12081](https://cloud.githubusercontent.com/assets/155205/23760646/b6750a8a-0533-11e7-9ffc-9463929770e9.gif)

**Comment by jsolisu** [comment]
Confirmed on version 1.17.0 dev.

**Comment by camillejr** [comment]
Hi, I had the same issue when trying to change the Line Height to 0.8 in the Editor Settings.

**Comment by t9md** [comment]
I think label `more-information-needed` and `needs-reproduction` is incorrect since it's reproducable now.

**Comment by nathansobo** [comment]
Is this still a problem on 1.19? I was deliberately delaying dealing with it in anticipation of all the code changing. If it's still a problem we should investigate.

**Comment by t9md** [comment]
@nathansobo

I checked.
Your anticipation was right.
I can not reproduce issue with v1.19, FIXED in v1.19-beta.

I can reproduce with v1.18(current stable), it need slight modification of reproduce code I pasted above.

I think we can close this issue.

Here is latest reproduce code, just for reference.
- Can reproduce v1.18-stable
- Can NOT reproduce v1.19-beta

```javascript
(function makeEditorGetBlankAfterSplitPane() {
  const pane = atom.workspace.getActivePane()
  const editor = atom.workspace.getActiveTextEditor()
  const editorElement = editor.element
  const scrollTop = editorElement.getScrollTop()

  atom.workspace.open("/tmp/file1", {split: 'right'}).then((file1) => {
    file1.onDidDestroy(() => {
      pane.activate()
      pane.activateItem(editor)
      editorElement.setScrollTop(scrollTop)
    })
    atom.workspace.open("/tmp/file2", {pane, activatePane: false})
  })
})()
// 1. Run code above(paste dev-console), result in left-pane(file2), right-pane(file1, active)
// 2. cmd-w to close file1, automatically close file2
// 3. So you are now at original editor before you run code.
// 4. cmd-k right to split right,
// 5. see left pane(original editor) get blanked clicking pane throw error.

```

**Comment by nathansobo** [comment]
Thanks very much for checking @t9md. I'm going to close this as fixed in 1.19 which is currently on beta.
